rosidl_generator_cpp/include/rosidl_generator_cpp/MessageTypeSupport.h

```diff
@@ -17,13 +17,22 @@
 #define __rosidl_generator_cpp__MessageTypeSupport__h__
 
 #include <rosidl_generator_c/message_type_support.h>
+#include <rosidl_generator_c/visibility_control.h>
 
 namespace rosidl_generator_cpp
 {
 
 template<typename T>
+ROSIDL_PUBLIC
 const rosidl_message_type_support_t * get_type_support_handle();
 
 }  // namespace rosidl_generator_cpp
 
+// Tail include the implementation for this function.
+// This include is provided by the implementation and therefore only
+// one version of this header will get included from the path.
+// This header maps the get_type_support_handle function to an implementation
+// specific version of that function.
+#include <rosidl_generator_cpp/get_type_support_handle_impl.hpp>
```

Comment:
Sorry, difficult to describe without real code.

I was suggesting to create the file `rosidl_generator_cpp/get_type_support_handle_impl.hpp` in this package and only include an `#error` directive letting the user know that it lacks a proper implementation of this and possibly point him to some packages which provide it.